aksk/pkg/iam/driver/keystone/keystone.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2021 The SODA Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keystone
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/opensds/multi-cloud/aksk/pkg/model"
+	"github.com/opensds/multi-cloud/aksk/pkg/utils"
+	pb "github.com/opensds/multi-cloud/aksk/proto"
+)
+
+const (
+	KEYSTONE_URI    = "/identity/v3/credentials"
+	PROTOCOL        = "http://"
+	AK_LENGTH       = 16
+	SK_LENGTH       = 32
+	POST            = "POST"
+	GET             = "GET"
+	DELETE          = "DELETE"
+	AUTH_TOKEN      = "X-Auth-Token"
+	CONTENT_TYPE    = "Content-Type"
+	APPL_JSON       = "application/json"
+	USER_QUERY_STR  = "?user_id="
+	SEPERATOR       = "/"
+	CREDENTIAL_TYPE = "ec2"
+)
+
+type KeystoneIam struct {
+	Host string
+	URI  string
+}
+
+type Client struct {
+}
+
+var Keystone = &KeystoneIam{}
+
+func Init(host string) *KeystoneIam {
+	Keystone.Host = host
+	return Keystone
+}
+
+func (iam *KeystoneIam) CreateAkSk(aksk *model.AkSk, req *pb.AkSkCreateRequest) (*model.AkSkOut, error) {
+	akey := utils.GenerateRandomString(AK_LENGTH)
+	skey := utils.GenerateRandomString(SK_LENGTH)
+
+	blb := &model.Blob{Access: akey, Secret: skey}
+	blbout, err := json.Marshal(blb)
+	if err != nil {
+		panic(err)
+	}
+
+	u, err := json.Marshal(model.Credential{Credential: model.CredBody{ProjectId: aksk.ProjectId,
+		UserId: aksk.UserId,
+		Blob:   string(blbout),
+		Type:   CREDENTIAL_TYPE}})
+
+	client := &http.Client{}
+	keystoneURL := PROTOCOL + iam.Host + KEYSTONE_URI
+	postreq, err := http.NewRequest(POST, keystoneURL, bytes.NewBuffer(u))
+	if err != nil {
+		return nil, err
+	}
+	postreq.Header.Add(AUTH_TOKEN, req.Token)
+	postreq.Header.Set(CONTENT_TYPE, APPL_JSON)
+
+	akskresp, _ := client.Do(postreq)
```

Comment:
IMO error check is must from returned http response

Reply:
done